src/mds/Mutation.h

```diff
@@ -86,30 +98,33 @@ struct MutationImpl : public TrackedOp {
       reserve(32);
     }
 
-    void add_rdlock(SimpleLock *lock) {
-      emplace_back(lock, LockOp::RDLOCK);
+    void add_rdlock(SimpleLock *lock, int idx=-1) {
+      add_lock(LockOp(lock, LockOp::RDLOCK), idx);
     }
     void erase_rdlock(SimpleLock *lock);
     void add_xlock(SimpleLock *lock, int idx=-1) {
-      if (idx >= 0)
-	emplace(cbegin() + idx, lock, LockOp::XLOCK);
-      else
-	emplace_back(lock, LockOp::XLOCK);
+      add_lock(LockOp(lock, LockOp::XLOCK), idx);
     }
     void add_wrlock(SimpleLock *lock, int idx=-1) {
-      if (idx >= 0)
-	emplace(cbegin() + idx, lock, LockOp::WRLOCK);
-      else
-	emplace_back(lock, LockOp::WRLOCK);
+      add_lock(LockOp(lock, LockOp::WRLOCK), idx);
     }
     void add_remote_wrlock(SimpleLock *lock, mds_rank_t rank) {
       ceph_assert(rank != MDS_RANK_NONE);
-      emplace_back(lock, LockOp::REMOTE_WRLOCK, rank);
+      add_lock(LockOp(lock, LockOp::REMOTE_WRLOCK, rank), -1);
     }
     void lock_scatter_gather(SimpleLock *lock) {
-      emplace_back(lock, LockOp::WRLOCK | LockOp::STATE_PIN);
+      add_lock(LockOp(lock, LockOp::WRLOCK | LockOp::STATE_PIN), -1);
     }
     void sort_and_merge();
+
+  protected:
+    void add_lock(LockOp op, int idx) {
```

Comment:
Suggestion: `insert_lock` will match the argument list and the behavior better

Reply:
Existing code uses `add_Zlock` (for some `Z`). I didn't want to change the verb just for the helper method.

Reply:
I looked at this method being used in the `acquire_locks` and it was confusing. I had to open the implementation to see what it does with the index. Back then I didn't realize you added this function in the PR, thought it was there already.

Reply:
>  I didn't want to change the verb just for the helper method.

But this method has different behavior and signature. I believe that warrants a matching name.

Reply:
> But this method has different behavior and signature. I believe that warrants a matching name.

Sorry I don't understand? The `add_lock` method is literally the same code the methods used to do but refactored.

Reply:
I'll drop this one. I didn't realize the scope of the refactoring to implement it following my suggestion.